libs/frontend/feature-site-store-init/src/lib/use-api-store.ts

```diff
@@ -150,17 +150,21 @@ export const useApiStore = (props: IUseApiStoreProps): ISiteStore => {
   // The API timer is reset if currently active
   const save = async (site: ISite, options?: ISiteSaveOptions): Promise<void> => {
     const storedSite = storeSite(site)
-    store.site.setSite(storedSite)
     let changed = false
     for (const key in site) {
       const k = key as keyof IStoredSiteDirty
-      dirty.value[k] = site[k] !== store.site[k]?.value
+      dirty.value[k] = dirty.value[k] || storedSite[k] !== store.site[k]?.value
```

Comment:
The updated data was being compared before it was serialized, so it was always marked as dirty!

I also realized that if some data changed previously, we shouldn't overwrite the `dirty` value, otherwise it won't be send in the update.

Reply:
Got it!
Thanks for all the detailed explanation!